### PR TITLE
Support OBJECT REFCOUNT

### PR DESCRIPTION
**Pull request by Avital-Fine**
https://redis.io/commands/object-refcount/

#2055

### CODE REVIEW

File: src/StackExchange.Redis/Interfaces/IDatabase.cs [hunk 1]
```diff
@@ -620,6 +620,15 @@ public interface IDatabase : IRedis, IDatabaseAsync
         /// <remarks>https://redis.io/commands/randomkey</remarks>
         RedisKey KeyRandom(CommandFlags flags = CommandFlags.None);
 
+        /// <summary>
+        /// Returns the reference count of the stored at <paramref name="key"/>.
+        /// </summary>
+        /// <param name="key">The key to rename.</param>
+        /// <param name="flags">The flags to use for this operation.</param>
+        /// <returns>The number of references.</returns>
+        /// <remarks>https://redis.io/commands/object-refcount/</remarks>
```
Comment (NickCraver) [inline src/StackExchange.Redis/Interfaces/IDatabase.cs:629]:
```suggestion
        /// <summary>
        /// Returns the reference count of the object stored at <paramref name="key"/>.
        /// </summary>
        /// <param name="key">The key to get a reference count for.</param>
        /// <param name="flags">The flags to use for this operation.</param>
        /// <returns>The number of references (0 if the key does not exist).</returns>
        /// <remarks>https://redis.io/commands/object-refcount</remarks>
```

File: src/StackExchange.Redis/Interfaces/IDatabaseAsync.cs [hunk 2]
```diff
@@ -596,6 +596,15 @@ public interface IDatabaseAsync : IRedisAsync
         /// <remarks>https://redis.io/commands/randomkey</remarks>
         Task<RedisKey> KeyRandomAsync(CommandFlags flags = CommandFlags.None);
 
+        /// <summary>
+        /// Returns the reference count of the stored at <paramref name="key"/>.
+        /// </summary>
+        /// <param name="key">The key to rename.</param>
+        /// <param name="flags">The flags to use for this operation.</param>
+        /// <returns>The number of references.</returns>
+        /// <remarks>https://redis.io/commands/object-refcount/</remarks>
```
Comment (NickCraver) [inline src/StackExchange.Redis/Interfaces/IDatabaseAsync.cs:605]:
```suggestion
        /// <summary>
        /// Returns the reference count of the object stored at <paramref name="key"/>.
        /// </summary>
        /// <param name="key">The key to get a reference count for.</param>
        /// <param name="flags">The flags to use for this operation.</param>
        /// <returns>The number of references (0 if the key does not exist).</returns>
        /// <remarks>https://redis.io/commands/object-refcount</remarks>
```

**Review by NickCraver — CHANGES_REQUESTED** [review]
This looks great - a few doc tweaks and good to go :+1: - since it's just docs I'll merge in and watch builds. Need to add release notes for this one yet!

**Comment by NickCraver** [comment]
I'll try and tweak this a bit - added a not-exist case locally and that'll throw because it's nil coming back from Redis. We can either make that 0, or `long?` - checking other methods.

**Review by NickCraver — APPROVED** [review]
Looking good - thanks again! :+1:

**Comment by NickCraver** [comment]
@Avital-Fine I changed the return here given what redis returns and consistency with a few other APIs for that case (null does clearly indicate missing) - can you confirm this makes sense to you? If not we can tweak further.

**Comment by Avital-Fine** [comment]
> @Avital-Fine I changed the return here given what redis returns and consistency with a few other APIs for that case (null does clearly indicate missing) - can you confirm this makes sense to you? If not we can tweak further.

Makes sense, but need to update the doc ;)

**Comment by Avital-Fine** [comment]
> This looks great - a few doc tweaks and good to go +1 - since it's just docs I'll merge in and watch builds. Need to add release notes for this one yet!

@NickCraver Sorry! Always forget the release notes :sweat_smile:

**Pull request merged**